src/mongo_logger.ts

```diff
@@ -522,22 +554,45 @@ export class MongoLogger {
       ...clientOptions,
       mongodbLogPath: resolveLogPath(envOptions, clientOptions)
     };
-    const defaultSeverity =
-      parseSeverityFromString(combinedOptions.MONGODB_LOG_ALL) ?? SeverityLevel.OFF;
+    const defaultSeverity = resolveSeverityConfiguration(
+      combinedOptions.mongodbLogComponentSeverities?.default,
+      combinedOptions.MONGODB_LOG_ALL,
+      SeverityLevel.OFF
+    );
 
     return {
       componentSeverities: {
-        command: parseSeverityFromString(combinedOptions.MONGODB_LOG_COMMAND) ?? defaultSeverity,
-        topology: parseSeverityFromString(combinedOptions.MONGODB_LOG_TOPOLOGY) ?? defaultSeverity,
-        serverSelection:
-          parseSeverityFromString(combinedOptions.MONGODB_LOG_SERVER_SELECTION) ?? defaultSeverity,
-        connection:
-          parseSeverityFromString(combinedOptions.MONGODB_LOG_CONNECTION) ?? defaultSeverity,
-        client: parseSeverityFromString(combinedOptions.MONGODB_LOG_CLIENT) ?? defaultSeverity,
+        command: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.command,
+          combinedOptions.MONGODB_LOG_COMMAND,
+          defaultSeverity
+        ),
+        topology: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.topology,
+          combinedOptions.MONGODB_LOG_TOPOLOGY,
+          defaultSeverity
+        ),
+        serverSelection: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.serverSelection,
+          combinedOptions.MONGODB_LOG_SERVER_SELECTION,
+          defaultSeverity
+        ),
+        connection: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.connection,
+          combinedOptions.MONGODB_LOG_CONNECTION,
+          defaultSeverity
+        ),
+        client: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.client,
+          combinedOptions.MONGODB_LOG_CLIENT,
+          defaultSeverity
+        ),
         default: defaultSeverity
       },
       maxDocumentLength:
-        parseUnsignedInteger(combinedOptions.MONGODB_LOG_MAX_DOCUMENT_LENGTH) ?? 1000,
+        parseUnsignedInteger(combinedOptions.mongodblogMaxDocumentLength) ??
```

Comment:
```suggestion
        combinedOptions.mongodbLogMaxDocumentLength ??
```

We can just have the `mongodbLogMaxDocumentLength` be a number on the client options so we can do that validation inside `parseOptions`.

Reply:
pushed changes!